pyteal/ast/global_.py

```diff
@@ -19,13 +19,13 @@ class GlobalField(Enum):
     logic_sig_version = (5, "LogicSigVersion", TealType.uint64, 2)
     round = (6, "Round", TealType.uint64, 2)
     latest_timestamp = (7, "LatestTimestamp", TealType.uint64, 2)
-    current_app_id = (8, "CurrentApplicationID", TealType.uint64, 2)
+    current_application_id = (8, "CurrentApplicationID", TealType.uint64, 2)
```

Comment:
I don't think we can change this without breaking anything tho... for `GlobalField` is exported in `__init__`, right?

Reply:
I did some experiment, and some conclusion drawn:
- `Enum` can have duplicate fields with different names and same value
- we do can deprecate a field in enum, by following example:
```python
class _DeprecateGlobalFieldMeta(EnumMeta):
    def __getattribute__(self, __name: str) -> Any:
        deprecated_fields_and_reasons: dict[str, str] = {"min_txn_fee": "reason"}
        if __name in deprecated_fields_and_reasons:
            warnings.warn(
                deprecated_fields_and_reasons[__name], DeprecationWarning, stacklevel=2
            )
        return EnumMeta.__getattribute__(self, __name)


class GlobalField(Enum, metaclass=_DeprecateGlobalFieldMeta):
```

so we can push through the change by deprecating old fields and use new fields internally.

Reply:
it is clever and TIL metaclass but I wonder if this particular naming inconsistency is worth the extra mechanism.

Reply:
The need is to align method names with enum fields:
- `current_application_id`
- `current_application_address`
- `caller_application_id`
- `caller_application_address`

While we can add new fields of these names into enum `GlobalField`, we want to deprecate fields with old names. It is still unclear to me how to deprecate a field in enum, until up to now with metaclass.